Summary filter

dodumosu opened the pull request:
this pull request replaces #948

dodumosu committed:
feat: implement exponents in QA

dodumosu committed:
feat: add filter for submission values

dodumosu committed:
chore: remove duplicated property

dodumosu committed:
feat: implement value search and hotlinking

this branch replaces the work on
nditech/apollo#948

takinbo reviewed (changes requested):
<img width="447" alt="Screenshot 2024-03-05 at 21 07 09" src="https://github.com/nditech/apollo/assets/41004/82c21c7d-1817-4d1b-9b08-1c03de5f9b97">

So I'm filling out the search box and I'm getting "No results found". I get confused because I don't know what that means; does it mean the question doesn't exist? I would prefer if the dropdown was a little more helpful and could make suggestions on what to fill out. Something similar to what we have for the QA builder:

<img width="568" alt="Screenshot 2024-03-05 at 21 09 39" src="https://github.com/nditech/apollo/assets/41004/7bd2900b-790e-499f-835b-91f653aeb655">

dodumosu committed:
feat: use expression builder for value search

dodumosu committed:
feat: use tom-select for first field dropdown

dodumosu committed:
feat: use custom dropdown on value

dodumosu committed:
chore: remove comment

dodumosu committed:
chore: switch to selectize from tom-select

dodumosu committed:
chore: remove tom-select artifacts

dodumosu committed:
chore: don't use label in dropdown

dodumosu committed:
fix: destroy selectize control when source changes to value

dodumosu committed:
chore: add destroyed hook

dodumosu committed:
feat: after changing source, warn for incomplete term

dodumosu committed:
feat: clean up selectize when term is removed.

dodumosu committed:
chore: various UI changes

- remove icon on button
- change button text color

dodumosu committed:
fix: don't destroy all the dropdowns

dodumosu committed:
fix: destroy all selectize instances on expression deletion

dodumosu committed:
fix: correct issue with locations import (#979)

dependabot[bot] committed:
chore(deps): bump gettext.js from 0.8.0 to 2.0.3 in /apollo/static (#978)

Bumps gettext.js from 0.8.0 to 2.0.3.

---
updated-dependencies:
- dependency-name: gettext.js
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dodumosu committed:
fix: fix PWA issues (#980)

this commit fixes issues reported from users.
the issues are due to upgrading marshmallow

dodumosu committed:
feat: start using new template

dodumosu committed:
feat: working with new template

dodumosu committed:
chore: change delete button visibility rule

this commit changes the delete term button display rule.
now, it displays on each term unless there's only one available

dodumosu committed:
chore: remove unused code

dependabot[bot] committed:
chore(deps-dev): bump webpack from 5.76.0 to 5.94.0 in /apollo/static (#982)

Bumps [webpack](https://github.com/webpack/webpack) from 5.76.0 to 5.94.0.
- [Release notes](https://github.com/webpack/webpack/releases)
- [Commits](https://github.com/webpack/webpack/compare/v5.76.0...v5.94.0)

---
updated-dependencies:
- dependency-name: webpack
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dodumosu committed:
fix: replace all usage of _joined_entities() (#981)

* fix: replace all usage of _joined_entities()

SQLAlchemy 1.4 removed an internal query method (_joined_entities())
that was used to check if a query was already joined to a model.
This pull request replaces the usages of that method.

* fix: add missing file

* feat: add unit tests and comment

takinbo committed:
fixed a few nits

takinbo reviewed (approved):
tACK
